### DISCUSSION TITLE
Scrape error, posts, 404

### DISCUSSION THREAD

**Discussion by karpatxnegen** [kevinzg/facebook-scraper#424] [General]
We have one scraped fb page, where the following error occurs:
`404 Client Error: Not Found for url: https://m.facebook.com/tasolaszlokepviselo/posts/ - [page_iterators.iter_pages(facebook_scraper.page_iterators) line:27]`

Anyone have an idea what can cause this?
There is no such problem with the other scraped pages.

**Comment by neon-ninja** [comment]
Some pages just don't have the /posts endpoint or the posts tab. The scraper should log the error and fallback to / in this case, extraction should still work

**Reply by karpatxnegen** [reply]
`6d7ed370b08f-fb-scraper: [INFO]: 2021-08-02 00:15:01.969Z  - Setting user agent: Mozilla/5.0 (Linux; Android 6.0.1; E6653 Build/32.2.A.0.253) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.98 Mobile Safari/537.36 - [main.scrape_group(root) line:167]
6d7ed370b08f-fb-scraper: [ERROR]: 2021-08-02 00:15:02.697Z  - Exception while requesting URL: https://m.facebook.com/tasolaszlokepviselo/posts/
6d7ed370b08f-fb-scraper: [ERROR]: 2021-08-02 00:15:02.697Z  - 404 Client Error: Not Found for url: https://m.facebook.com/tasolaszlokepviselo/posts/ - [page_iterators.iter_pages(facebook_scraper.page_iterators) line:27]
6d7ed370b08f-fb-scraper: [WARNING]: 2021-08-02 00:15:05.080Z  - No raw posts (<article> elements) were found in this page. - [page_iterators._get_page(facebook_scraper.page_iterators) line:181]
6d7ed370b08f-fb-scraper: [INFO]: 2021-08-02 00:15:05.082Z  - Page parser did not find next page URL - [page_iterators.generic_iter_pages(facebook_scraper.page_iterators) line:95]`

I'm sorry to bothering you, but I don't understand, why and how this page differs from others, where the scraper works very well

**Reply by neon-ninja** [reply]
Do you get any locale warnings?

**Reply by karpatxnegen** [reply]
`
[ERROR]: 2021-08-11 08:50:03.052Z  - Exception while requesting URL: https://m.facebook.com/tasolaszlokepviselo/posts/
Exception: HTTPError('404 Client Error: Not Found for url: https://m.facebook.com/tasolaszlokepviselo/posts/') - [facebook_scraper.get(facebook_scraper.facebook_scraper) line:480]
Traceback (most recent call last):
  File "~/.local/lib/python3.8/site-packages/facebook_scraper/facebook_scraper.py", line 431, in get
    response.raise_for_status()
  File "~/.local/lib/python3.8/site-packages/requests/models.py", line 953, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://m.facebook.com/tasolaszlokepviselo/posts/

[ERROR]: 2021-08-11 08:50:03.053Z  - 404 Client Error: Not Found for url: https://m.facebook.com/tasolaszlokepviselo/posts/ - [page_iterators.iter_pages(facebook_scraper.page_iterators) line:27]

~/.local/lib/python3.8/site-packages/facebook_scraper/facebook_scraper.py:422: UserWarning: Locale detected as hu_HU - for best results, set to en_US
  warnings.warn(f"Locale detected as {locale} - for best results, set to en_US")

[WARNING]: 2021-08-11 08:50:04.775Z  - No raw posts (<article> elements) were found in this page. - [page_iterators._get_page(facebook_scraper.page_iterators) line:180]

[INFO]: 2021-08-11 08:50:04.775Z  - Page parser did not find next page URL - [page_iterators.generic_iter_pages(facebook_scraper.page_iterators) line:95]
`
that's all what I get :(

**Reply by karpatxnegen** [reply]
As I learned now, this page is empty: https://m.facebook.com/tasolaszlokepviselo, without login (as the scraper sees it, because we use a proxy).
But there are posts, if we are trying the desktop version (whithout m. ).
I don't know if this is a setting on that page, or a bug.

**Reply by neon-ninja** [reply]
It's not a bug, it's just one of the differences between desktop and mobile. See https://github.com/kevinzg/facebook-scraper/issues/412#issuecomment-889474016
